### PR TITLE
Tariff reallocator utility

### PR DESCRIPTION
**Pull request by iaincrawf**
No npm updates yet, will try for them in a separate pr

### CODE REVIEW

File: src/Facade/ResourceBuilders/SalesArticlesReallocatorResourceBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    using Linn.Products.Resources;
+
+    public class SalesArticlesReallocatorResourceBuilder : IResourceBuilder<ResponseModel<SalesArticlesReallocator>>
+    {
+        public SalesArticlesReallocatorResource Build(ResponseModel<SalesArticlesReallocator> salesArticlesReallocator)
+        {
+            return new SalesArticlesReallocatorResource
+            {
+                OldTariffId = salesArticlesReallocator.ResponseData.OldTariffId.ToString(),
+                NewTariffId = salesArticlesReallocator.ResponseData.NewTariffId.ToString(),
```
Comment (richardiphillips) [inline src/Facade/ResourceBuilders/SalesArticlesReallocatorResourceBuilder.cs:17]: why are you making these strings out of interest?

File: src/Facade/Services/SalesArticleService.cs [hunk 2]
```diff
@@ -1,19 +1,25 @@
 ﻿namespace Linn.Products.Facade.Services
 {
     using System;
+    using System.Collections.Generic;
     using System.Linq;
     using System.Linq.Expressions;
+    using System.Security.Cryptography.X509Certificates;
```
Comment (richardiphillips) [inline src/Facade/Services/SalesArticleService.cs:7]: is this doing anything?

File: src/Facade/Services/SalesArticleService.cs [hunk 3]
```diff
@@ -22,6 +28,32 @@ public class SalesArticleService : FacadeService<SalesArticle, string, SalesArti
             : base(repository, transactionManager)
         {
             this.coreTypeRepository = coreTypeRepository;
+            this.salesArticleRepository = repository;
+            this.transactionManager = transactionManager;
+        }
+
+        public IResult<ResponseModel<SalesArticlesReallocator>> Reallocate(string oldTariffId, string newTariffId, IEnumerable<string> privileges)
+        {
+            var oldId = int.Parse(oldTariffId);
```
Comment (richardiphillips) [inline src/Facade/Services/SalesArticleService.cs:37]: could still be ints at this point?

File: src/Facade/Services/SalesArticleService.cs [hunk 4]
```diff
@@ -22,6 +28,32 @@ public class SalesArticleService : FacadeService<SalesArticle, string, SalesArti
...
+            var newId = int.Parse(newTariffId);
+
+            try
+            {
+                var articlesForReallocation = this.salesArticleRepository.FindAll().Where(x => x.TariffId == oldId);
+
+                foreach (var salesArticle in articlesForReallocation)
+                {
+                    salesArticle.TariffId = newId;
```
Comment (richardiphillips) [inline src/Facade/Services/SalesArticleService.cs:46]: cool but testable logic so should be domain code

**Review by richardiphillips — APPROVED** [review]
nice one. good js testing.

File: src/Facade/ResourceBuilders/SalesArticlesReallocatorResourceBuilder.cs (hunk 1 above)
Comment (iaincrawf) [reply src/Facade/ResourceBuilders/SalesArticlesReallocatorResourceBuilder.cs:17]: Ah good point, they could all be ints now I think. I initially made a mistake and was using the wrong id which was a string, I seem to have got confused and left some as strings

File: src/Facade/Services/SalesArticleService.cs (hunk 2 above)
Comment (iaincrawf) [reply src/Facade/Services/SalesArticleService.cs:7]: Nope, no idea what I was trying that used that but removed now

File: src/Facade/Services/SalesArticleService.cs (hunk 4 above)
Comment (iaincrawf) [reply src/Facade/Services/SalesArticleService.cs:46]: Ahhh yeah sorry I forgot we don't put logic like that in facades, was just trying to get that into my head before furlough. Will move it

**Pull request merged**